docs/Integrating Your HMI/Developing the HMI UI/index.md

```diff
@@ -157,6 +157,14 @@ There are some minor customization options available for the app menu. An HMI ca
 
 ![Tiles Menu](./assets/tiles_menu.png)
 
+### Dynamic Menu Updating
+
+SDL enables the ability to dynamically load menu items and icons to improve system performance. In some cases an app may submit a large number of menu commands, sub menus, and icons. Processing these assets can use a large amount of system resources. To help mitigate performance issues, the HMI can choose when to request the app updates SDL Core with the missing menu contents.
```

Comment:
`the HMI can choose when to request the app updates SDL Core with the missing menu contents`
this part is confusing, maybe something like this sounds better?
`the HMI can choose when to request resources, at which time the app can update SDL Core with the missing menu contents.`